internal/config/config.go

```diff
@@ -167,16 +167,39 @@ func IsGroupsValid(groups []GroupConfig) bool {
 }
 
 type GlobalConfig struct {
-	Cache CacheConfig `yaml:"cache"`
-	Port  int         `yaml:"port"`
+	Cache   CacheConfig   `yaml:"cache"`
+	Routing RoutingConfig `yaml:"routing"`
+	Port    int           `yaml:"port"`
 }
 
 type CacheConfig struct {
 	Redis string `yaml:"redis"`
 }
 
+type ErrorsConfig struct {
+	HTTPCodes    []string `yaml:"httpCodes"`
+	JSONRPCCodes []string `yaml:"jsonRpcCodes"`
+	ErrorStrings []string `yaml:"errorStrings"`
+	Rate         float64  `yaml:"rate"`
+}
+
+type MethodConfig struct {
+	Name      string        `yaml:"method"`
+	Threshold time.Duration `yaml:"threshold"`
+}
+
+type LatencyConfig struct {
+	Methods   []MethodConfig `yaml:"methods"`
+	Threshold time.Duration  `yaml:"threshold"`
+}
+
 type RoutingConfig struct {
-	MaxBlocksBehind int `yaml:"maxBlocksBehind"`
+	AlwaysRoute     *bool         `yaml:"alwaysRoute"`
+	Errors          ErrorsConfig  `yaml:"errors"`
```

Comment:
WDYT about making `ErrorsConfig` and `LatencyConfig` pointers? 

With a non-pointer type, it's hard to tell if the struct was set or not by the user since the default value is a valid struct (with all its fields set to zero values). 

With a pointer type, the field is initialized to `nil` if the user doesn't supply the field. This makes it easier to check if Error/Latency based routing has been configured. e.g. `== nil` is easier than `(recursively) checking all the fields in the struct to be zero`. 

Admittedly, we do the first approach already with Cache config (see [here](https://github.com/satsuma-xyz/node-gateway/blob/2bde95011a028033d7b579340daeff07735e8a16/internal/config/config.go#L206) and [here](https://github.com/satsuma-xyz/node-gateway/blob/51c61567667450e73e92bd3d6e7bf89e6d06f3aa/internal/route/request_executor.go#L123)), but IMO I don't think it's ideal.